projects/plugins/social/src/class-rest-controller.php

```diff
@@ -62,15 +62,58 @@ public function require_admin_privilege_callback() {
 		);
 	}
 
+	/**
+	 * A list of all settings that can be retrieved and updated..
+	 *
+	 * Items for each setting:
+	 * - key: Key to be posted in the request body.
+	 * - get_callback: Function to get the current value of the setting.
+	 * - update_callback: Function to call to update the setting.
+	 * - validation_callback: Gets the return value of the update_callback.
+	 *     Return a boolean to indicate if the setting updated successfully.
+	 *
+	 * @return array
+	 */
+	public function get_available_settings() {
+		return array(
+			array(
+				'key'                 => 'publicize_active',
+				'get_callback'        => function () {
+					return ( new Modules() )->is_active( \Jetpack_Social::JETPACK_PUBLICIZE_MODULE_SLUG );
+				},
+				'update_callback'     => function ( $value ) {
+					return ( new Modules() )->update_status( \Jetpack_Social::JETPACK_PUBLICIZE_MODULE_SLUG, (bool) $value, false, false );
+				},
+				'validation_callback' => function ( $return ) {
+					return ! is_wp_error( $return );
+				},
+			),
+			array(
+				'key'                 => 'show_pricing_page',
+				'get_callback'        => function () {
+					return Jetpack_Social::should_show_pricing_page();
+				},
```

Comment:
As this is a callable, could it be?

```suggestion
				'get_callback'        => array( 'Jetpack_Social', 'should_show_pricing_page' ),
```

If we carry on using callbacks, the others could be defined in a similar way.